lib/fund.js

```diff
@@ -52,96 +53,52 @@ function printJSON (fundingInfo) {
+    const deps = flatCache[url]
 
-  // ---
-
-  const getFundingItems = (fundingItems) =>
-    Object.keys(fundingItems || {}).map((fundingItemName) => {
-      // first-level loop, prepare the pretty-printed formatted data
-      const fundingItem = fundingItems[fundingItemName]
-      const { version, funding } = fundingItem
-      const { type, url } = funding || {}
+    const packages = deps.map((dep) => {
+      const { name, version } = dep
 
       const printableVersion = version ? `@${version}` : ''
-      const printableType = type && { label: `type: ${funding.type}` }
-      const printableUrl = url && { label: `url: ${funding.url}` }
-      const result = {
-        fundingItem,
-        label: fundingItemName + printableVersion,
-        nodes: []
-      }
-
-      if (printableType) {
-        result.nodes.push(printableType)
-      }
-
-      if (printableUrl) {
-        result.nodes.push(printableUrl)
-      }
-
-      setStackedItem(funding, result)
-
-      return result
-    }).reduce((res, result) => {
-      // recurse and exclude nodes that are going to be stacked together
-      const { fundingItem } = result
-      const { dependencies, funding } = fundingItem
-      const items = getFundingItems(dependencies)
-      const stackedResult = retrieveStackedItem(funding)
-      items.forEach(i => result.nodes.push(i))
-
-      if (stackedResult && stackedResult !== result) {
-        stackedResult.label += `, ${result.label}`
-        items.forEach(i => stackedResult.nodes.push(i))
-        return res
-      }
-
-      res.push(result)
-
-      return res
-    }, [])
-
-  const [ result ] = getFundingItems({
-    [fundingInfo.name]: {
-      dependencies: fundingInfo.dependencies,
-      funding: fundingInfo.funding,
-      version: fundingInfo.version
+      return `${name}${printableVersion}`
+    })
+
+    return {
+      label: url,
+      nodes: [packages.join(', ')]
     }
   })
 
-  return archy(result, '', { unicode: opts.unicode })
+  return archy({ label: `${name}${printableVersion}`, nodes: items }, '', { unicode: opts.unicode })
 }
 
-function openFundingUrl (packageName, cb) {
+function openFundingUrl (packageName, fundingSourceNumber, cb) {
   function getUrlAndOpen (packageMetadata) {
     const { funding } = packageMetadata
-    const { type, url } = retrieveFunding(funding) || {}
-    const noFundingError =
-      new Error(`No funding method available for: ${packageName}`)
-    noFundingError.code = 'ENOFUND'
-    const typePrefix = type ? `${type} funding` : 'Funding'
-    const msg = `${typePrefix} available at the following URL`
-
-    if (validFundingUrl(funding)) {
+    const validSources = [].concat(retrieveFunding(funding)).filter(validFundingField)
+
+    if (validSources.length === 1 || (fundingSourceNumber > 0 && fundingSourceNumber <= validSources.length)) {
+      const { type, url } = validSources[fundingSourceNumber ? fundingSourceNumber - 1 : 0]
```

Comment:
not really a blocker but wouldn't most js devs expect an index such as `fundingSourceNumber` (and its respective `which` user-exposed config) to be 0-based? 🤔 I think I would and this could def confuse me 😝

Reply:
oh nvm, after seeing the output from the tap-snapshot I realize it's way less confusing since the indexes get printed along with each url 😊